client/src/components/ElementEditor/ElementEditor.js

```diff
@@ -80,17 +79,24 @@ class ElementEditor extends PureComponent {
       elementTypes,
       isDraggingOver,
       connectDropTarget,
+      allowedElements,
     } = this.props;
     const { dragTargetElementId, dragSpot } = this.state;
 
-    const classNames = classnames('element-editor', {
-      // 'element-editor--dragging': isDragging,
-    });
+    // Map the allowed elements because we want to retain the sort order provided by that array.
+    // We default to allowing all elements if the allowed elements array is not defined for
+    // backwards compatibility.
+    const allowedElementTypes =
+      Array.isArray(allowedElements)
```

Comment:
It'll probably always be an array but because the prop is not required it could be `undefined`.

Reply:
I don't think it will ever be undefined actually. On the php side it will always exist (and defaults to all sub-classes of BaseElement if allowed_elements is not set) so it will always get passed as a prop to the ElementArea component.